tests/components/layout/test_match.py

```diff
@@ -0,0 +1,245 @@
+        ("random", rx.text("fourth value")),
+        ({"foo": "bar"}, rx.text("fifth value")),
+        (MatchState.num + 1, rx.text("sixth value")),
+        rx.text("default value"),
+    )
+    match_comp = Match.create(MatchState.value, *match_case_tuples)
+    match_dict = match_comp.render()  # type: ignore
+    assert match_dict["name"] == "Fragment"
+
+    [match_child] = match_dict["children"]
+
+    assert match_child["name"] == "match"
+    assert str(match_child["cond"]) == "{match_state.value}"
+
+    match_cases = match_child["match_cases"]
+    assert len(match_cases) == 6
+
+    assert match_cases[0][0]._var_name == "1"
+    assert match_cases[0][0]._var_type == int
+    first_return_value_render = match_cases[0][1].render()
+    assert first_return_value_render["name"] == "Text"
+    assert first_return_value_render["children"][0]["contents"] == "{`first value`}"
+
+    assert match_cases[1][0]._var_name == "2"
+    assert match_cases[1][0]._var_type == int
+    assert match_cases[1][1]._var_name == "3"
+    assert match_cases[1][1]._var_type == int
+    second_return_value_render = match_cases[1][2].render()
+    assert second_return_value_render["name"] == "Text"
+    assert second_return_value_render["children"][0]["contents"] == "{`second value`}"
+
+    assert match_cases[2][0]._var_name == "[1, 2]"
+    assert match_cases[2][0]._var_type == list
+    third_return_value_render = match_cases[2][1].render()
+    assert third_return_value_render["name"] == "Text"
+    assert third_return_value_render["children"][0]["contents"] == "{`third value`}"
+
+    assert match_cases[3][0]._var_name == "random"
+    assert match_cases[3][0]._var_type == str
+    fourth_return_value_render = match_cases[3][1].render()
+    assert fourth_return_value_render["name"] == "Text"
+    assert fourth_return_value_render["children"][0]["contents"] == "{`fourth value`}"
+
+    assert match_cases[4][0]._var_name == '{"foo": "bar"}'
+    assert match_cases[4][0]._var_type == dict
+    fifth_return_value_render = match_cases[4][1].render()
+    assert fifth_return_value_render["name"] == "Text"
+    assert fifth_return_value_render["children"][0]["contents"] == "{`fifth value`}"
+
+    assert match_cases[5][0]._var_name == "(match_state.num + 1)"
+    assert match_cases[5][0]._var_type == int
+    fifth_return_value_render = match_cases[5][1].render()
+    assert fifth_return_value_render["name"] == "Text"
+    assert fifth_return_value_render["children"][0]["contents"] == "{`sixth value`}"
+
+    default = match_child["default"].render()
+
+    assert default["name"] == "Text"
+    assert default["children"][0]["contents"] == "{`default value`}"
+
+
+def test_match_vars():
+    """Test matching cases with return values as Vars."""
+    match_case_tuples = (
+        (1, "first"),
+        (2, 3, "second value"),
+        ([1, 2], "third-value"),
+        ("random", "fourth_value"),
+        ({"foo": "bar"}, "fifth value"),
```

Comment:
i think at least some of the return values should be stateful Var and we should make sure that the resulting match code includes the necessary `useContext` to pull in access to the given substate.